RationaleMCP/0031/differences.md

```diff
@@ -1234,3 +1234,81 @@ Accordingly, a function component declaration which is neither input nor output
+
+In Flat Modelica, every base-clock and sub-clock is declared as a component with a name.
+Sometimes, this name will correspond to the name of a clock in full Modelica, sometimes it will be an automatically generated name.
+Tools may find it useful to include the clock components in a simulation result, but doing so is not required and there is no standard for what to store.
+
+Instead of the binary clocked `sample` operator in full Modelica, Flat Modelica has an unary `sample(…)` operator.
+It is only allowed inside the equations and algorithms of a `subpartition`, and the semantics is that the argument expression is sampled at the clock ticks of the current sub-partition.
+
+A Flat Modelica model with clock partitioning can look like this:
+```
+package 'M'
+model 'M'
+  Real 'x';
+  Real 'baseVar', 'cVar1', 'cVar2', 'cVar3';
+  Real 'mixedVar1';
+
+/* Equations and algorithms before the start of the first base-partition belong to the continuous-time partition. */
+equation
+  der('x') = 1;
+
+partition /* Beginning of base-partition */
+  Clock 'baseClock' = Clock(1); /* Clock name originating from full Modelica model. */
+
+  subpartition /* Beginning of sub-partition within base-partition. */
+    Clock 'myClock' = 'baseClock';
+  equation
+    'baseVar' = sample('x');
+
+  subpartition
+    annotation(solverMethod = "ImplicitEuler");
+    Clock _subClock0 = subSample('baseClock', 2); /* Automatically generated clock name. */
+  equation
+    der('cVar1') = noClock('baseVar');
+
+  subpartition
+    Clock _subClock1 = superSample(subSample('baseClock', 2), 8);
+  equation
+    'cVar2' = noClock('baseVar');
+    'cVar3' = noClock('cVar1');
+  algorithm
+    'mixedVar1' := 'cVar2' + 'cVar3';
+
+partition
+  Clock _baseClock0 = Clock(1.1);
+  ...
+  /* Base-partition ends at the start of another base-partition, or at the end of the model. */
+
+end 'M';
+end 'M';
+```
+
+The base-clock of a `partition` must be given by a `Clock`-expression without references to other clocks.
+The clock of a `subpartition` must be given by a (clock conversion) expression of the current base-clock, and there may be more than one `subpartition` having the same relation to the base-clock.
+
+The solver method for discretized continuous-time equations is given by a `solverMethod` specification after the `subpartition`'s clock-clause.
+It is only required when the sub-partition contains continuous-time equations.
+
+In the equations and algorithms of a `subupartition`, references to variables from the continuous-time partition must appear inside the Flat Modelica unary `sample(…)` operator.
+Similarly, references to variables from another sub-partition must appear inside the `noClock(…)` or `previous(…)` operators.
+It is not allowed to reference variables determined in another clocked base-partition.
```

Comment:
Proposed change:
```suggestion
It is not allowed to reference variables determined in another clocked base-partition.
Except, by going through the continuous-time partition using `sample(hold(x))`.
```
Or possibly in parenthesis to indicate that it isn't that important.
Note: A variant would be to not allow this construct directly, but require that it is done using an extra variable (in the continuous-time partition).

Reply:
```suggestion
It is not allowed to reference variables determined in another clocked base-partition, except when wrapped in `hold()`.
(The expression `hold(x)` is a continuous-time expression and needs to be sampled before it can appear in a clocked partition.)
```